core/src/test/scala/unit/kafka/utils/ZkUtilsTest.scala

```diff
@@ -107,4 +123,27 @@ class ZkUtilsTest extends ZooKeeperTestHarness {
     assertEquals(None, zkUtils.getLeaderIsrAndEpochForPartition(topic, partition + 1))
   }
 
+
+  @Test
+  def testParsePartitionReassignmentDataWithoutDedup_invalidJson() = {
+    val jsonStr = "{invalid json}"
+
+    try {
+      ZkUtils.parsePartitionReassignmentData(jsonStr)
+      fail("Should have thrown ConfigException");
+    } catch {
+      case e: ConfigException =>
+        assertTrue(e.getMessage.contains("Invalid reassignment config"))
+    }
+  }
+
+  @Test
+  def testParsePartitionReassignmentData() = {
```

Comment:
We should rewrite these tests to use `ReassignPartitionsZNode`. Also, since they don't require `ZooKeeper`, they should be unit tests. Perhaps `ReassignPartitionsZNodeTest`.